lib/util.c

```diff
@@ -147,3 +154,87 @@ rc_strlcpy(char *dst, char const *src, size_t siz)
 }
 
 #endif
+
+#ifdef __linux__
+/** Set the network namespace for the current thread (or process - if single threaded).
+ *
+ * @param net_namespace - New network namespace to set.
+ * @param prev_ns_handle - Handle to previous network namespace.
+ *
+ * @return 0 on success, -1 when failure.
+ */
+int rc_set_netns(const char *net_namespace, int *prev_ns_handle)
+{
+    int rc = -1;
```

Comment:
the `ifdef` could go here, with an `#else` clause which always returns an error.